docs/website/docs/walkthroughs/deploy-a-pipeline/deploy-with-google-cloud-functions.md

```diff
@@ -0,0 +1,115 @@
+
+6. Further, you need to modify,  the `__init__.py` file in the notion folder, and modify the `notion_databases` function as follows:
+    
+    ```python
+    @dlt.source
+    def notion_databases(
+        database_ids: Optional[List[Dict[str, str]]] = None,
+        api_key = notion_apikey_env, # Refers to the notion secret
+    ) -> Iterator[DltResource]:
+    ```
+    
+    Note how we have referenced api_key to the variable named "notion_apikey_env" instead of "dlt.secrets.value".
+    ### Managing secrets with Secret Manager
+     Next, go to the Google Secrets Manager. In Google Secrets Manager, create the secret "Notion API Key", name the secret "notion_secret" and add "Notion API key" to the secret values. To learn more about Google Secrets Manager, read the full **[documentation](https://cloud.google.com/secret-manager/docs/create-secret-quickstart) here**.
+    
+      - Assign the "Secret Manager Secret Accessor" role to the service account that was used to create the function (this is usually the default service account associated with the Google Project where the function will be created).
+       - There would be a *default* *service account* associated with the project ID you are using, please assign the *Cloud Functions Developer* role to the associated service account. To learn more about the service account, read the [documentation](https://cloud.google.com/iam/docs/service-account-overview) here.
+
+## 2. Deploying GCP Cloud Function
+In a shell editor, navigate to the directory where the "main.py" file is located and run the following command in the terminal
+```bash
+  gcloud functions deploy pipeline_notion --runtime python310 --trigger-http --allow-unauthenticated --source .
+```
+        
+- This command deploys a function named "pipeline_notion" with Python 3.10 as the runtime environment, an HTTP trigger, and allows unauthenticated access. The source "." refers to all files in the directory.
+- See [the documentation](https://cloud.google.com/functions/docs) for more details on Google Cloud Functions.
+  
+  ### Setting up environmental variables in the Cloud function using Secret Manager
+  
+  After deploying the function, you need to create an environment variable in Cloud Function using the Notion Secret you created earlier.
+  Go to the Google Cloud function, click on the function you deployed earlier using the shell editor and select "EDIT".
+   1. In the 'Runtime, Build, Connections and Security Settings' section, select 'Security and Images Repo'.
+   2. Click on 'Add a secret reference' and select the secret you created, e.g. 'notion_secret'.
+   3. Set the 'Reference method' to 'Mounted as environment variable'.
+   4. In the 'Environment Variable' field, enter the name of the environment variable assigned in the source method, in this case,    'notion_apikey_env', and set the version to 'latest'.
```

Comment:
yes the pipeline ran, and I did like api_key = notion_apikey_env, but that could have been avoided if we made api_key as env variable. :) Missed that .  but pipeline ran and here is the link 

https://us-central1-dlt-dev-personal.cloudfunctions.net/mainpipelinenotion6

Reply:
would be great if we can include dlt reading from google secret manager

Reply:
is it in one of our gcp projects or this is your private account?

Reply:
This is a private account.

Reply:
is there any way I can experiment on it? otherwise I'd need to reproduce the whole thing. @adrianbr maybe we should build our demos/projects always in our infrastructure?

Reply:
For this time u can use my creds if comfortable, from next time. we'll build in dlt infra. If okay I'll send the creds on slack.

Reply:
redeploying in dlt-dev-external :))